Domain/Shop.java

```diff
@@ -1,3 +1,47 @@
+import java.util.HashMap;
+import java.util.Map;
+
 public class Shop {
-    
+    private Integer m_ShopId;
+    private String m_ShopFounder;
+    private Map<Integer ,Product> m_ProductMap; // <ProductId, Product> 
```

Comment:
You will have a problem when trying to add an item to the shop. maybe add the quantity of the item?